go-controller/pkg/util/node_annotations.go

```diff
@@ -312,6 +315,39 @@ func NodeChassisIDAnnotationChanged(oldNode, newNode *kapi.Node) bool {
 	return oldNode.Annotations[ovnNodeChassisID] != newNode.Annotations[ovnNodeChassisID]
 }
 
+type ManagementPortDetails struct {
+	PfId   int `json:"PfId"`
+	FuncId int `json:"FuncId"`
+}
+
+func SetNodeManagementPortAnnotation(nodeAnnotator kube.Annotator, PfId int, FuncId int) error {
+	mgmtPortDetails := ManagementPortDetails{
+		PfId:   PfId,
+		FuncId: FuncId,
+	}
+	bytes, err := json.Marshal(mgmtPortDetails)
+	if err != nil {
+		return fmt.Errorf("failed marshaling")
+	}
+	return nodeAnnotator.Set(ovnNodeManagementPort, string(bytes))
+}
+
+// ParseNodeManagementPort returns the parsed host addresses living on a node
+func ParseNodeManagementPortAnnotation(node *kapi.Node) (int, int, error) {
+	mgmtPortAnnotation, ok := node.Annotations[ovnNodeManagementPort]
+	if !ok {
+		return -1, -1, newAnnotationNotSetError("%s annotation not found for node %q", ovnNodeManagementPort, node.Name)
+	}
+
+	cfg := ManagementPortDetails{}
+	if err := json.Unmarshal([]byte(mgmtPortAnnotation), &cfg); err != nil {
+		return -1, -1, fmt.Errorf("failed to unmarshal management port annotation %s for node %q: %v",
+			mgmtPortAnnotation, node.Name, err)
+	}
+
+	return cfg.PfId, cfg.FuncId, nil
```

Comment:
I'm thinking, why function return `int, int` and not `ManagementPortDetails` itself? Just asking, not requesting to change it.

Reply:
@DmytroLinkin Are you satisfied with my reply? I'd like to close this conversation as resolved or move forward to fixing it if you like to see something else.

Reply:
Let's keep as is. I was evaluating what need to be done here, so later in the future SF support for DPU/DPUHost modes will introduce less changes.